File: Solutions/200. Number of Islands/200.py
class Solution:
    def numIslands(self, grid: List[List[str]]) -> int:
        n = len(grid)
        m = len(grid[0])
        ans = 0
        def makeZero( i: int, j: int):
            if i<0 or j<0 or i==n or j==m:
                return 
            if grid[i][j] == '1':
                grid[i][j] = 0
                makeZero( i-1, j)
                makeZero( i, j-1)
                makeZero( i, j+1)
                makeZero( i+1, j)

        for i in range(n):
            for j in range(m):
                if grid[i][j] == '1':
                    makeZero( i, j)
                    ans += 1
        return ans

File: Solutions/200. Number of Islands/200.java
class Solution {
    public int numIslands(char[][] grid) {
        int n=grid.length, m = grid[0].length;
        int ans = 0;
        for(int i=0;i<n;i++){
            for(int j=0;j<m;j++){
                if(grid[i][j] == '1'){
                    makeZero(grid, i, j);
                    ans++;
                }
            }
        }
        return ans;
    }
    public void makeZero(char[][] grid, int i, int j){
        int n=grid.length, m = grid[0].length;
        if(i<0 || j<0 || i==n || j==m){
            return;
        }
        if(grid[i][j] == '1'){
            grid[i][j] = '0';
            makeZero(grid, i-1, j);
            makeZero(grid, i+1, j);
            makeZero(grid, i, j-1);
            makeZero(grid, i, j+1);
        }
    }
}